packages/media-utils/src/utils/upload-media.ts

```diff
@@ -67,7 +70,18 @@ export function uploadMedia( {
 	onError,
 	onFileChange,
 	signal,
+	multiple = true,
 }: UploadMediaArgs ) {
+	if ( ! multiple && filesList.length > 1 ) {
```

Comment:
Let's add a unit test for this accordingly in `packages/media-utils/src/utils/test/upload-media.ts`